src/node.cc

```diff
@@ -3235,7 +3239,12 @@ void SetupProcessObject(Environment* env,
       process_env_template->NewInstance(env->context()).ToLocalChecked();
   process->Set(FIXED_ONE_BYTE_STRING(env->isolate(), "env"), process_env);
 
+#ifdef __CloudABI__
+  READONLY_PROPERTY(process, "pid",
+                    String::NewFromUtf8(env->isolate(), program_getpid()));
```

Comment:
This should be a number, not a string, for consistency.

Reply:
Sure. One thing we should prevent, though, is truncation/rounding of such a number. Would you happen to know how I could store a 128 bit number that way?

Reply:
[BigInt](https://github.com/tc39/proposal-bigint) support will be coming soon to JavaScript as a language feature, which IMHO would be the most appropriate type to use. This discussion about how to handle >53 bit integers has been had on the issue tracker at least a couple of times now and I believe BigInt is the correct way forward.

BigInt is currently at stage 3, which is the [second-to-last stage](https://tc39.github.io/process-document/) before being submitted to ECMA.

Reply:
Awesome! That sounds like a pretty good fit. Let's not expose it as a string then.

Shall I remove this part from the pull request and only leave the changes to `src/env.cc`?

Reply:
I'm not sure that will work as-is because if your `getpid()` implementation is returning a string, POSIX branches are expecting an integer value still and you will get compiler errors.